### PR TITLE
adds api route for all articles. adds passing test for happy path

### PR DESCRIPTION
**Pull request by osaxon**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -21,3 +21,24 @@ describe("/api/topics", () => {
...
 });
+
+describe("/api/articles", () => {
+    test("GET:200 responds with an array of all articles sorted sorted by date in descending order.", () => {
+        return request(app)
+            .get("/api/articles")
+            .expect(200)
+            .then(({ body }) => {
+                const { articles } = body;
+                articles.forEach((article) => {
```
Comment (Keb84) [inline __tests__/app.test.js:86]: if asserting inside a forEach you should check the length of the array first. What would happen if the response was an empty array? 🤔

File: __tests__/app.test.js [hunk 2]
```diff
@@ -21,3 +21,24 @@ describe("/api/topics", () => {
...
+                    expect(article.title).toEqual(expect.any(String));
+                    expect(article.article_id).toEqual(expect.any(Number));
+                    expect(article.topic).toEqual(expect.any(String));
+                    expect(article.created_at).toEqual(expect.any(String));
+                    expect(article.votes).toEqual(expect.any(Number));
+                    expect(article.article_img_url).toEqual(expect.any(String));
+                    expect(article.comment_count).toEqual(expect.any(Number));
+                });
+            });
+    });
```
Comment (Keb84) [inline __tests__/app.test.js:97]: have you checked for if the date is sorted correctly?

**Review by Keb84 — CHANGES_REQUESTED** [review]
Please have a look at my comments and make the necessary changes and resubmit 👍

**Comment by osaxon** [comment]
Made changes to check articles are sorted by created date. updated the SQL query for selecting articles as it was wrong previously (changes to LEFT OUTER JOIN)

**Review by Keb84 — APPROVED** [review]
Nicely done 👍 merge when ready 🎸

**Pull request merged**